driver/src/orderbook/streamed/state.rs

```diff
@@ -323,17 +323,122 @@ mod tests {
+                amount: U256::from($amount),
+            };
+            $state = $state.apply_event(&Event::Withdraw(event), $batch).unwrap();
+        };
+        (to $state:ident in batch $batch:expr, place order number $index:expr, from user $user:expr, selling $sell_amount:expr, of token $sell_token:expr, for at least $buy_amount:expr, of token $buy_token:expr, in batch interval [$valid_from:expr, $valid_until:expr]) => {
+            let event = OrderPlacement {
+                owner: address($user),
+                index: $index,
+                buy_token: $buy_token,
+                sell_token: $sell_token,
+                valid_from: $valid_from,
+                valid_until: $valid_until,
+                price_numerator: $sell_amount,
+                price_denominator: $buy_amount,
+            };
+            $state = $state
+                .apply_event(&Event::OrderPlacement(event), $batch)
+                .unwrap();
+        };
+        (to $state:ident in batch $batch:expr, trade order number $index:expr, from user $user:expr, selling $sell_amount:expr, for $buy_amount:expr) => {
+            let event = Trade {
+                owner: address($user),
+                order_id: $index,
+                executed_sell_amount: $sell_amount,
+                executed_buy_amount: $buy_amount,
+                ..Default::default()
+            };
+            $state = $state.apply_event(&Event::Trade(event), $batch).unwrap();
+        };
+        (to $state:ident in batch $batch:expr, revert trade order number $index:expr, from user $user:expr, selling $sell_amount:expr, for $buy_amount:expr) => {
+            let event = TradeReversion {
+                owner: address($user),
+                order_id: $index,
+                executed_sell_amount: $sell_amount,
+                executed_buy_amount: $buy_amount,
+                ..Default::default()
+            };
+            $state = $state
+                .apply_event(&Event::TradeReversion(event), $batch)
+                .unwrap();
+        };
+        (to $state:ident in batch $batch:expr, submit solution from user $user:expr, with fee $fee:expr) => {
+            let event = SolutionSubmission {
+                submitter: address($user),
+                burnt_fees: U256::from($fee),
+                ..Default::default()
+            };
+            $state = $state
+                .apply_event(&Event::SolutionSubmission(event), $batch)
+                .unwrap();
+        };
+    }
+
+    macro_rules! assert_balance {
+        (in $state:ident batch $batch:expr, user $user:expr, has token $token:expr, balance $balance:expr) => {
+            assert_eq!(
+                account_state(&$state, $batch).read_balance($token, address($user)),
+                U256::from($balance)
+            );
+        };
+    }
+
+    macro_rules! assert_used_amount {
+        (in $state:ident batch $batch:expr, of order number $index:expr, from user $user:expr, is $amount:expr) => {
```

Comment:
ubernit:
```suggestion
        (in $state:ident for batch $batch:expr; of order number $index:expr, from user $user:expr, is $amount:expr) => {
```

Same as above